src/Decoder.php

```diff
@@ -110,7 +110,8 @@ public function handleData($data)
             // this looks like a multiline value, so only remember offset and wait for next newline
             $last = \substr(\end($data), -1);
             \reset($data);
-            if ($last === "\n" && ($newline === 1 || $this->buffer[$newline - 1] !== $this->enclosure)) {
+	    $edgeCase = \substr($this->buffer, $newline - 2, 3);
+            if ($last === "\n" && ($newline === 1 || $this->buffer[$newline - 1] !== $this->enclosure || $edgeCase === $this->delimiter . $this->enclosure ."\n")) {
```

Comment:
Thanks for looking into this :+1: 
Your casing seems a bit off, just a minor fix?

The rest looks fine to me :)